zsh-system-clipboard.zsh

```diff
@@ -8,132 +8,100 @@
-				} else {
-					suggest_to_install 'Termux:API (from Play Store), termux-api (from apt package)'
-				}
-				;;
-
-			linux*|freebsd*)
-				if (hash xclip 2>/dev/null) {
-					local clipboard_selection
-
-					case $ZSH_SYSTEM_CLIPBOARD_XCLIP_SELECTION {
-						PRIMARY)
-							clipboard_selection='PRIMARY'
-							;;
-
-						CLIPBOARD)
-							clipboard_selection='CLIPBOARD'
-							;;
-
-						*)
-							if [[ $ZSH_SYSTEM_CLIPBOARD_XCLIP_SELECTION != '' ]] {
-								error "\033[01m$ZSH_SYSTEM_CLIPBOARD_XCLIP_SELECTION\033[0m is not a valid value for \$ZSH_SYSTEM_CLIPBOARD_XCLIP_SELECTION. Please assign either 'PRIMARY' or 'CLIPBOARD'."
-
-							} else {
-								clipboard_selection='CLIPBOARD'
-							}
-							;;
-					}
+function error() {
+	echo -e "\n\n  \033[41;37m ERROR \033[0m \033[01mzsh-system-clipboard:\033[0m $@\n" >&2
+}
 
-					typeset -g CLIPBOARD[set]="xclip -sel $clipboard_selection -in"
-					typeset -g CLIPBOARD[get]="xclip -sel $clipboard_selection -out"
-				} else {
-					suggest_to_install 'xclip'
-				}
-				;;
+function suggest_to_install() {
+	error "Could not find any available clipboard manager. Make sure you have \033[01m${@}\033[0m installed."
+}
 
-			*)
-				error 'Unsupported system.'
-				;;
+declare -A ZSH_SYSTEM_CLIPBOARD
+case "$OSTYPE" {
+	darwin*)
+		if ((hash pbcopy && hash pbpaste) 2>/dev/null) {
+			typeset -g ZSH_SYSTEM_CLIPBOARD[set]='pbcopy'
+			typeset -g ZSH_SYSTEM_CLIPBOARD[get]='pbpaste'
+		} else {
+			suggest_to_install 'pbcopy, pbpaste'
 		}
-	}
-
-	function sub_set() {
-		if [[ "$ZSH_SYSTEM_CLIPBOARD_TMUX_SUPPORT" != '' ]] {
-			# Set also tmux clipboard buffer if tmux available.
-			if (hash tmux &>/dev/null && [[ "$TMUX" != '' ]]) {
-				tmux set-buffer -- "$@"
+		;;
+	linux-android*)
+		if ((hash termux-clipboard-set && hash termux-clipboard-get) 2>/dev/null) {
+			typeset -g ZSH_SYSTEM_CLIPBOARD[set]='termux-clipboard-set'
+			typeset -g ZSH_SYSTEM_CLIPBOARD[get]='termux-clipboard-get'
+		} else {
+			suggest_to_install 'Termux:API (from Play Store), termux-api (from apt package)'
+		}
+		;;
+	linux*|freebsd*)
+		if (hash xclip 2>/dev/null) {
+			local clipboard_selection
+			case $ZSH_SYSTEM_CLIPBOARD_XCLIP_SELECTION {
+				PRIMARY)
+					clipboard_selection='PRIMARY'
+					;;
+				CLIPBOARD)
+					clipboard_selection='CLIPBOARD'
+					;;
+				*)
+					if [[ $ZSH_SYSTEM_CLIPBOARD_XCLIP_SELECTION != '' ]] {
+						error "\033[01m$ZSH_SYSTEM_CLIPBOARD_XCLIP_SELECTION\033[0m is not a valid value for \$ZSH_SYSTEM_CLIPBOARD_XCLIP_SELECTION. Please assign either 'PRIMARY' or 'CLIPBOARD'."
+					} else {
+						clipboard_selection='CLIPBOARD'
+					}
+					;;
 			}
+			typeset -g ZSH_SYSTEM_CLIPBOARD[set]="xclip -sel $clipboard_selection -in"
+			typeset -g ZSH_SYSTEM_CLIPBOARD[get]="xclip -sel $clipboard_selection -out"
+		} else {
+			suggest_to_install 'xclip'
+		}
+		;;
+	*)
+		error 'Unsupported system.'
+		;;
+}
+unfunction error
```

Comment:
The thing is, if user already has a function named as `error` before sourcing the plugin, that function will be gone forever, even if you `unfunction` them properly.